### PR TITLE
Add a --volume flag to `modal shell`

### PR DESCRIPTION
**Pull request by mwaskom**
Supports `modal shell --volume ...` to quickly attach one (or more) of your volumes to an add-hoc shell for filesystem exploration:

```
$ modal shell --volume example-models
root@modal:/# ls /mnt/example-models
gpt5.pt  gpt6.pt
```

You can call it multiple times, like `modal shell --volume models --volume data`

Volumes are always mounted at `/mnt/{name}`. I considered making this configurable, e.g. `modal shell --volume /mnt/data:my-data-vol`. But that felt like overkill for something that targets quick work. If we find a need for it, I think it's fine to expand the interface in the future to allow it. We're reasonably strict about volume names and I think any valid volume name would also be a valid mount point? (Open to counter-arguments here !).

- Closes MOD-2404

## Changelog

- Volumes can now be mounted to an ad hoc modal shell session:
    
    ```
    modal shell --volume my-vol-name
    ```
    When the shell starts, the volume will be mounted at `/mnt/my-vol-name`. This may be helpful for shell-based exploration or manipulation of volume contents.

    Note that the option can be used multiple times to mount additional models:
    ```
    modal shell --volume models --volume data
    ```

### CODE REVIEW

**Review by ekzhang — APPROVED** [review]

File: modal/cli/run.py [hunk 1]
```diff
@@ -341,6 +345,13 @@ def shell(
         default=None, help="Container image tag for inside the shell (if not using FUNC_REF)."
     ),
     add_python: Optional[str] = typer.Option(default=None, help="Add Python to the image (if not using FUNC_REF)."),
+    volume: Optional[typing.List[str]] = typer.Option(
+        default=None,
+        help=(
+            "Name of a `modal.Volume` to mount inside the shell at `/mnt/{name}` (if not using FUNC_REF)."
```
Comment (ekzhang) [inline modal/cli/run.py:351]: Inside the _container_ perhaps?

**Comment by charlesfrye** [comment]
¿como se `llamas`?

Screenshot 2024-08-02 at 1 27 25 PM

This is great. I agree on the auto-mounting.

**Review by charlesfrye — COMMENTED** [review]
lgtm

**Comment by erikbern** [comment]
nice!

**Pull request merged**